packages/lesswrong/server/scripts/makeMigrations.ts

```diff
@@ -55,6 +56,234 @@ const schemaFileHeaderTemplate = `-- GENERATED FILE
 --
 `
 
+declare global {
+  type SchemaDependency =
+    {type: "extension", name: PostgresExtension} |
+    {type: "collection", name: CollectionNameString} |
+    {type: "function", name: string} |
+    {type: "view", name: string};
+}
+
+abstract class Node {
```

Comment:
We build a graph of "things" to put in the schema, where each separate "thing" inherits from `Node`